src/antsibull/build_changelog.py

```diff
@@ -69,64 +112,117 @@ def optimize_release_entry(entry: ChangelogGeneratorEntry) -> ChangelogGenerator
             for line in value.splitlines():
                 column_widths[i] = max(column_widths[i], len(line))
+    return column_widths
+
+
+def render_rst_table(headings: list[str], cells: list[list[str]]) -> str:
+    column_widths = compute_column_widths(headings, cells)
+    lines: list[str] = []
+    _add_table_line(lines, column_widths, "-", "-", "-", "+")
+    _add_table_row(lines, column_widths, headings, "|")
+    _add_table_line(lines, column_widths, "=", "=", "=", "+")
+    for row in cells:
+        _add_table_row(lines, column_widths, row, "|")
+        _add_table_line(lines, column_widths, "-", "-", "-", "+")
+    return "\n".join(lines)
+
 
-    # Add rows
-    _add_rst_table_line(builder, column_widths, "-")
-    _add_rst_table_row(builder, column_widths, headings)
-    _add_rst_table_line(builder, column_widths, "=")
+def render_md_table(headings: list[str], cells: list[list[str]]) -> str:
+    column_widths = compute_column_widths(headings, cells)
+    lines: list[str] = []
+    _add_table_row(lines, column_widths, headings, "|")
+    _add_table_line(lines, column_widths, " ", "-", " ", "|")
     for row in cells:
-        _add_rst_table_row(builder, column_widths, row)
-        _add_rst_table_line(builder, column_widths, "-")
+        _add_table_row(lines, column_widths, row, "|")
+    return "\n".join(lines)
+
+
+def format_link(title: str, url: str, text_format: TextFormat) -> str:
+    if text_format == TextFormat.RESTRUCTURED_TEXT:
+        return f"`{title} <{url}>`_"
+    if text_format == TextFormat.MARKDOWN:
+        return f"`[{title}]({url})"
+    raise ValueError(f"Unknown format {format}")
+
+
+def render_table(
+    renderer: AbstractRenderer,
+    headings: list[str],
+    cells: list[list[str]],
+    text_format: TextFormat,
+) -> None:
+    if text_format == TextFormat.RESTRUCTURED_TEXT:
+        renderer.add_text(render_rst_table(headings, cells), text_format=text_format)
+        return
+    if text_format == TextFormat.MARKDOWN:
+        renderer.add_text(render_md_table(headings, cells), text_format=text_format)
+        return
+    raise ValueError(f"Unknown format {text_format}")
 
 
 def append_changelog_changes_collections(
-    builder: RstBuilder,
+    renderer: AbstractRenderer,
     collection_metadata: CollectionsMetadata,
     changelog_entry: ChangelogEntry,
     is_last: bool,
+    text_format: TextFormat,
 ) -> PluginDataT:
     result: PluginDataT = []
 
     if changelog_entry.changed_collections:
-        builder.add_section(
-            "Included Collections" if is_last else "Changed Collections", 1
+        section = renderer.add_section(
+            "Included Collections" if is_last else "Changed Collections"
         )
-        builder.add_raw_rst(
+        section.add_text(
             "If not mentioned explicitly, the changes are reported in the combined changelog"
-            " below.\n"
+            " below.",
+            text_format=TextFormat.RESTRUCTURED_TEXT,
```

Comment:
Why isn't this

```suggestion
            text_format=text_format,
```

?

Reply:
For this piece of text it doesn't really matter, but generally I either want to use something like `TextFormat.PLAINTEXT` (which isn't supported yet) or specify another explicit format which definitely matches (in this case, `TextFormat.RESTRUCTURED_TEXT`) to avoid potential problems with conflicting markup in other formats added in the future.